llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -1165,21 +1144,24 @@ NVPTXTargetLowering::LowerGlobalAddress(SDValue Op, SelectionDAG &DAG) const {
 
 std::string NVPTXTargetLowering::getPrototype(
     const DataLayout &DL, Type *retTy, const ArgListTy &Args,
-    const SmallVectorImpl<ISD::OutputArg> &Outs, MaybeAlign retAlignment,
-    std::optional<std::pair<unsigned, const APInt &>> VAInfo,
-    const CallBase &CB, unsigned UniqueCallSite) const {
+    const SmallVectorImpl<ISD::OutputArg> &Outs, MaybeAlign RetAlign,
+    std::optional<std::pair<unsigned, unsigned>> VAInfo, const CallBase &CB,
```

Comment:
The second pair parameter is actually uint64_t now.

Reply:
That's an implementation detail, I think. It makes little difference here, but in general, I'd rather pass the value through as is without assuming how the value is supposed to be populated somewhere else. 

Considering that a fair number  of `getPrototype` inputs are derived from `CLI` and `VADeclareParam`, perhaps we should just pass the references to them to this function, and retrieve relevant parts directly. It would obviate the need for std::optional, for example.

Reply:
Okay, I just went ahead and removed this parameter completely. I don't think it makes sense to pull this value back out of the DAG when we know already what it is going to be. This is also consistent with how we do things in AsmPrinter.

https://github.com/llvm/llvm-project/blob/dd2a1590c34d250b70de6d9a4a11b9a13aacb8ac/llvm/lib/Target/NVPTX/NVPTXAsmPrinter.cpp#L1448-L1451

Reply:
Nice. I think it works better than passing bits and pieces separately.